ui/src/app/views/notebook-list/component.html

```diff
@@ -18,7 +18,40 @@ <h3 class="notebooks-header">Notebooks</h3>
   </div>
 
   <div class="card item-card notebook-card" *ngFor="let notebook of notebookList">
-    <div class="name" [class.disabled]="createDisabled" (click)="createDisabled || openNotebook(notebook)">{{notebook.name}}</div>
+    <div class="header-area">
+      <clr-dropdown [clrCloseMenuOnItemClick]="false">
+        <button type="button" class="notebook-menu" [disabled]="actionsDisabled" clrDropdownTrigger>
+          <clr-icon shape="ellipsis-vertical" class="ellipsis-vertical" [class.disabled]="actionsDisabled"></clr-icon>
+        </button>
+        <clr-dropdown-menu *clrIfOpen>
+          <button class="action-item"
+                  id="rename-{{notebook.name.replace('.ipynb', '')}}"
+                  (click)="renameThis(notebook)"
+                  [class.disabled]="actionsDisabled"
+                  [disabled]="actionsDisabled">
+            <clr-icon shape="pencil"></clr-icon>
+            Rename
+          </button>
+          <button class="action-item"
+                  id="clone-{{notebook.name.replace('.ipynb', '')}}"
```

Comment:
I'm not sure about this pattern, there can be pretty arbitrary characters in notebook names including spaces which are not generally valid as ids.

Reply:
I'd agree, but this id is for testing. When we have control over the names being used.

Reply:
Since this in the real template, I'm not sure I'm comfortable chalking it up to only needing to work for test purposes. Does this not cause errors when you have a notebook with a space in the name?

Also do you actually need this for your test - can you not just grab the appropriate button via class name after opening the menu? Are the menus for all notebooks rendered but hidden for some reason?

Reply:
So no errors when you have a notebook with spaces (or other special characters that I've tested). I'll try rewriting the test without this pattern.

Reply:
I've now also tried removing the pattern and grabbing the button via class name, and that does not seem to work. I'll continue playing around with it. But I'm leaning heavily towards leaving this as is.

Reply:
OK, I don't feel strongly enough to push further on this.